jax/experimental/ode.py

```diff
@@ -92,7 +92,7 @@ def initial_step_size(fun, t0, y0, order, rtol, atol, f0):
 
 def runge_kutta_step(func, y0, f0, t0, dt):
   # Dopri5 Butcher tableaux
-  alpha = jnp.array([1 / 5, 3 / 10, 4 / 5, 8 / 9, 1., 1., 0], dtype=f0.dtype)
+  alpha = jnp.array([1 / 5, 3 / 10, 4 / 5, 8 / 9, 1., 1., 0], dtype=t0.dtype)
```

Comment:
`alpha` is used only in l112, where it is combined with the two times `t0` and `dt`. It should have the same dtype as the times, rather than the return value `f0`.

Reply:
Good catch – though what happens if someone inadvertently passes in an integer sequence of times? Should we promote `t0` and `dt` to inexact somewhere?

Reply:
Good point - if an integer `t0` ever makes it here, `alpha` would be wrong if using `t0.dtype`!
Safer would be `f0.real.dtype` or `_complex_elem_type(f0)` from `jax._src.numpy.lax_numpy`?

Right now, `odeint(func=lambda x, t: x, y0=jnp.array(1.), t=jnp.array([0, 1]))` fails elsewhere, so if we want to make that run then `t` should probably be promoted further outside - in `_odeint` maybe? I don't know enough about jax types to do that unfortunately.

Reply:
Makes sense – I think the easiest fix then would be to (somewhere) add `assert np.issubdtype(t, np.floating)`. The `odeint` docstring says that `t` is an array of floats, so there's probably not any good reason to handle other cases.